src/backend/distributed/utils/citus_copyfuncs.c

```diff
@@ -73,6 +74,20 @@ CitusSetTag(Node *node, int tag)
 	} \
 	while (0)
 
+#define COPY_STRING_LIST(fldname) \
+	char *curString = NULL; \
+	List *newList = NIL; \
+	foreach_ptr(curString, from->fldname) { \
+		char *newString = curString ? pstrdup(curString) : (char *) NULL; \
+		if (newList == NIL) { \
+			newList = list_make1(newString); \
+		} \
+		else { \
+			newList = lappend(newList, newString); \
+		} \
```

Comment:
Doesn't appending to `NIL` create a new list that contains only appended element? Do you need this branching?

Reply:
I guess it does:

```c
List *
lappend(List *list, void *datum)
{
	Assert(IsPointerList(list));

	if (list == NIL)
		list = new_list(T_List);
	else
		new_tail_cell(list);

	lfirst(list->tail) = datum;
	check_list_invariants(list);
	return list;
}
```